### DISCUSSION TITLE
How do i make this?

### DISCUSSION THREAD

**Discussion by aimankhairy** [eritislami/evobot#253] [Q&A] [answered]
i want the searching command to be like this. you can get the song via reactions (no need to type 1-10) (like the bot senpai)
How do i do it? TQ

**Comment by brandonst** [comment, marked as answer]
First replace the message collector with a reaction collector, check which one the user reacted with, then play the song. 
Only replace 
```js
module.exports = {
    ...
}
```

With this
```js
module.exports = {
    name: "search",
    description: "Search and select videos to play",
    async execute(message, args) {
        if (!args.length)
            return message.reply(`Usage: ${message.client.prefix}${module.exports.name} <Video Name>`).catch(console.error);
        if (message.channel.activeCollector)
            return message.reply("A message collector is already active in this channel.");
        if (!message.member.voice.channel)
            return message.reply("You need to join a voice channel first!").catch(console.error);

        const search = args.join(" ");

        let resultsEmbed = new MessageEmbed()
            .setTitle(`**Reply with the song number you want to play**`)
            .setDescription(`Results for: ${search}`)
            .setColor("#F8AA2A");

        const results = await youtube.searchVideos(search, 10);
        if (!results.length) return message.channel.send('No results found for that search!')
        results.forEach((video, index) => resultsEmbed.addField(video.shortURL, `${index + 1}. ${video.title}`));

        const resultsMessage = await message.channel.send(resultsEmbed);
        const emojis = ['1️⃣', '2️⃣', '3️⃣', '4️⃣', '5️⃣', '6️⃣', '7️⃣', '8️⃣', '9️⃣', '🔟']
        emojis.forEach(async e => await resultsMessage.react(e).catch(() => { }))
        const filter = (reaction, user) => user.id === message.author.id && emojis.includes(reaction.emoji.name)

        message.channel.activeCollector = true;
        const response = await resultsMessage.awaitReactions(filter, { max: 1, time: 60000 });
        if (!response.first()) return message.channel.send('Time limit exceeded, the search has been cancelled.')
        const choice = results[parseInt(emojis.indexOf(response.first().emoji.name))].url;

        message.channel.activeCollector = false;
        message.client.commands.get("play").execute(message, [choice]);
        resultsMessage.delete().catch(() => { });
    }
};
```

**Reply by aimankhairy** [reply]
tq

**Reply by kasepta** [reply]
i got bug, when search timeout, and try to search again. got this message
`A message collector is already active in this channel.`
@brandonst

**Reply by mihawkman** [reply]
same problem here @brandonst

**Reply by brandonst** [reply]
Heres the fixed version, my apologizes
```js
module.exports = {
    name: "search",
    description: "Search and select videos to play",
    async execute(message, args) {
        if (!args.length)
            return message.reply(`Usage: ${message.client.prefix}${module.exports.name} <Video Name>`).catch(console.error);
        if (message.channel.activeCollector)
            return message.reply("A message collector is already active in this channel.");
        if (!message.member.voice.channel)
            return message.reply("You need to join a voice channel first!").catch(console.error);

        const search = args.join(" ");

        let resultsEmbed = new MessageEmbed()
            .setTitle(`**Reply with the song number you want to play**`)
            .setDescription(`Results for: ${search}`)
            .setColor("#F8AA2A");

        const results = await youtube.searchVideos(search, 10);
        if (!results.length) return message.channel.send('No results found for that search!')
        results.forEach((video, index) => resultsEmbed.addField(video.shortURL, `${index + 1}. ${video.title}`));

        const resultsMessage = await message.channel.send(resultsEmbed);
        const emojis = ['1️⃣', '2️⃣', '3️⃣', '4️⃣', '5️⃣', '6️⃣', '7️⃣', '8️⃣', '9️⃣', '🔟']
        emojis.forEach(async e => await resultsMessage.react(e).catch(() => { }))
        const filter = (reaction, user) => user.id === message.author.id && emojis.includes(reaction.emoji.name)

        message.channel.activeCollector = true;
        const response = await resultsMessage.awaitReactions(filter, { max: 1, time: 60000 });
        if (!response.first()) {
            return message.channel.send('Time limit exceeded, the search has been cancelled.')
            message.channel.activeCollector = false;
        }
        const choice = results[parseInt(emojis.indexOf(response.first().emoji.name))].url;

        message.channel.activeCollector = false;
        message.client.commands.get("play").execute(message, [choice]);
        resultsMessage.delete().catch(() => { });
    }
};
```

**Reply by mihawkman** [reply]
> ```js
> module.exports = {
>     name: "search",
>     description: "Search and select videos to play",
>     async execute(message, args) {
>         if (!args.length)
>             return message.reply(`Usage: ${message.client.prefix}${module.exports.name} <Video Name>`).catch(console.error);
>         if (message.channel.activeCollector)
>             return message.reply("A message collector is already active in this channel.");
>         if (!message.member.voice.channel)
>             return message.reply("You need to join a voice channel first!").catch(console.error);
> 
>         const search = args.join(" ");
> 
>         let resultsEmbed = new MessageEmbed()
>             .setTitle(`**Reply with the song number you want to play**`)
>             .setDescription(`Results for: ${search}`)
>             .setColor("#F8AA2A");
> 
>         const results = await youtube.searchVideos(search, 10);
>         if (!results.length) return message.channel.send('No results found for that search!')
>         results.forEach((video, index) => resultsEmbed.addField(video.shortURL, `${index + 1}. ${video.title}`));
> 
>         const resultsMessage = await message.channel.send(resultsEmbed);
>         const emojis = ['1️⃣', '2️⃣', '3️⃣', '4️⃣', '5️⃣', '6️⃣', '7️⃣', '8️⃣', '9️⃣', '🔟']
>         emojis.forEach(async e => await resultsMessage.react(e).catch(() => { }))
>         const filter = (reaction, user) => user.id === message.author.id && emojis.includes(reaction.emoji.name)
> 
>         message.channel.activeCollector = true;
>         const response = await resultsMessage.awaitReactions(filter, { max: 1, time: 60000 });
>         if (!response.first()) {
>             return message.channel.send('Time limit exceeded, the search has been cancelled.')
>             message.channel.activeCollector = false;
>         }
>         const choice = results[parseInt(emojis.indexOf(response.first().emoji.name))].url;
> 
>         message.channel.activeCollector = false;
>         message.client.commands.get("play").execute(message, [choice]);
>         resultsMessage.delete().catch(() => { });
>     }
> };
> ```
when it says Time limit exceeded, the search has been cancelled it still shows the error when i try to search other songs

**Reply by brandonst** [reply]
Another mistake i made, in this part
```js
        if (!response.first()) {
            return message.channel.send('Time limit exceeded, the search has been cancelled.')
            message.channel.activeCollector = false;
        }
```

replace it with this
```js
        if (!response.first()) {
            message.channel.activeCollector = false;
            return message.channel.send('Time limit exceeded, the search has been cancelled.')
        }
```

**Reply by BlazeIsClone** [reply]
Works like a charm nice work Thank you, xD

**Comment by aka-pal** [comment]
```
            return message.channel.send('Time limit exceeded, the search has been cancelled.')
```
Remove this line from it's current position amd this line after
```
message.channel.activeCollector = false;
```
It should look like below
```
        const response = await resultsMessage.awaitReactions(filter, { max: 1, time: 60000 });
        if (!response.first()) {
            message.channel.activeCollector = false;
            return message.channel.send('Time limit exceeded, the search has been cancelled.');
        }
```

**Reply by aka-pal** [reply]
For me it's working, try yourself

**Reply by mihawkman** [reply]
its working now Tq

**Comment by karimawi** [comment]
@brandonst Hey, do you know a way that I can have both commands like the deafault `search` and then the original command `masssearch` or `msearch` because it lets me choose multiple results from the search... do you know a way I can do this?

or you may have both methods running, like I can choose from the reactions or type to choose multiple tracks to play, this would be **way better**
